---
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Worksheet 1 - welcome to R

Welcome to the second part of the Bioinformatics class! 

As you know, the first part of the class focused on Perl, the command line and emacs. This second part of the class will focus on R. It is taught by me, Daniel Fulop. We'll do R for 5 weeks and after that, you have another 3.5 weeks to work on your projects. 

In this class, you will be mostly working by yourself or with a partner. You get worksheets and work at your own pace. Worksheets are due 1 week after they're assigned in class. If you turn them in late, you will be deducted 10% for each day late. I am working on a means of digital submission of worksheets ~~probably through iLearn~~ through GitHub.

**Task 1** - Be a class ambassador and visit my office hours. 

**Goal:** Each day of the course (except day 1), I would like to know how you are doing. I also would like to get to know everyone. Each of you should therefore sign up to come to my office hours once during the coming 5 weeks. My office hours are Tuesday and Thursday 1:30-2:30. Use this sign up spreadsheet: https://goo.gl/4rw9i6


When you come to my office hours (probably with a colleague), be prepared to tell me at least one thing you found interesting in the last class and one thing you found confusing, hard or boring. Office hours will be held in this computer lab.

**Did you sign up for an office hour meeting?**

Yes, November 15 (Maria del Pilar lopez)


Name: Maria del Pilar lopez


Date: November 15 

***  
#Worksheet 2 - R inspiration

R is a software environment and programming language for statistics that is used by many people, among them many biologists. The goal of this worksheet is for you to find inspiration to learn R by reading about how other people use R and how these people got started with R. 

####Task 1 - read four stories from https://whyiuser.wordpress.com/
Go to the website, pick a story that looks like it could be interesting and answer the following questions. Repeat four times.

**Story #1:**

1. What is the R user's name? Stephen Piccolo

2. What topic do they work on? Cancer Genomics

3. Which answer did you find most surprising / what did you learn from reading their story?
I like the fact that he says not to read too much about R, but to start practicing it. I also appreciate he added his scripts for this project at:
https://github.com/srp33/TCGA_RNASeq_Clinical/tree/master/Scripts

http://bioinformatics.oxfordjournals.org/content/31/22/3666


**Story #2:**

1. What is the R user's name? Robert M Flight

2. What topic do they work on? Bioinformatics of multi-omics

3. Which answer did you find most surprising / what did you learn from reading their story?  

I like how he integrated several aspects in  his Analysis Package:
Vignettes providing the full calculations of the correlation of nucleosome associated PARP1 reads in the MCF-7 and MDA-MB123 cell lines with
â€¢	CTCF ChIP-Seq
â€¢	Methylation reads
â€¢	Histone Marks ChIP-Seq
â€¢	Transcript Expression
And what I like the most, is that is that the scripts and data are available at:
https://github.com/rmflight/fmcorrelationbreastcaparp1

**Story #3:**

1. What is the R user's name? Victor Garcia

2. What topic do they work on? Evolutionary Dynamics and Epidemics

3. Which answer did you find most surprising / what did you learn from reading their story?  
I like that he said that it was hard to start working with R and he suggested the R Inferno tutorial. I already downloaded it. 


**Story #4:**

1. What is the R user's name? Anthony Greenberg

2. What topic do they work on? Quantitative Genetics

3. Which answer did you find most surprising / what did you learn from reading their story?  

Vectorize everything and avoid loops like the plague.


####Task 2 - summarize

After reading several stories about people who use R, is there anything problem that you think R can help you to solve? Is there anything particular that caught your interest?

I would like to do RNA seq with R to help my lab design microRNA experiment (picking the right morpholinos according to the published SRA data for Xenopus laevis). I think that I will take a look at Piccoloâs RNA seq scripts to see if I understand how to adjust them to my lab needs


Is there anything that didn't make sense to you or was unexpected? 
It seems that only few are computer engineers.


####Task 3 - share with your neighbor
Compare notes with at least one neighbor. Why did you neighbor choose the stories he or she read? Did you pick up something different? Did the stories change your ideas about R?
More or less the same.


**Finished!** 

Name: Pilar Lopez

Date:  

***

#Worksheet 3 - Do chapter 1 of online R tutorial https://www.codeschool.com/courses/try-r

You need to create a login for Codeschool, then you get started with the R class.  
For this worksheet, you'll do chapter 1. 

1. Do you see any differences with Perl? Write down at least 3 differences
Perl starts counting at zero while R starts counting in 1
Perl does not have a nice window interface like R does
I have a pc, and I was not able to run Perl in my pc unless I was signed into the cluster. R is great because I can use it with my pc.

2. How would you ask R to multiply 7 and 8? Or to divide 7 by 8? 

```
7 * 8
7 / 8
```

3. Which of the following expressions would return a logical (or boolean) value? 

* A. `3 < 4`       TRUE returns logical value
* B. `3 == 4`      FALSE  retruns logical value   
* C. `x <- 4`
* D. `4 -> x`

4. Can you assign a logical value to a variable called "y"? How would you assign FALSE to a variable called "y"?

```{bash, eval=FALSE}
Y <- FALSE
```

5. Name three functions you learned in the tutorial. What do they do? 
Addition 
```{r example}
example<-c(1, 2, 3, 4)
sum(example)
```
Repetition

```{r}
rep("hi", times = 3)
```
Takes the square root 

```{r}
sqrt (16)
```



6. What is the function you use to list all the files in your current directory? 


```{r}
list.files()
```

7. What is the function you use to run all the code in a file called "file.R"? 


```{bash, eval=FALSE}
source("file.R")

```

8. Do you have any questions about the tutorial? 
Not really, this is pretty good. Thanks. 

**Finished!**

Name: Pilar Lopez


Date:  

***

#Worksheet 4 - Do chapter 2 of online R tutorial http://tryr.codeschool.com/

Login to Codeschool and do chapter 2.

1. Define the following data structures? What do you find written in them?

Boolean: these are logical propositions such TRUE or FALSE

Numeric: numbers

String : characters


2. What happens if you try to combine in a vector a Boolean, a numeric and a string? 
Vectors cannot hold values with different modes (types). All the values were converted to a single mode (characters) so that the vector can hold them all.

3. What would be the result of this piece of code: `seq(1, 2, 0.1)`  

```
 seq(1, 2, 0.1) 
 ```
 
 [1] 1.0 1.1 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 2.0

4. A vector can have names. How do you call an element of a vector by its name

```
Nameofthevector <-c(1, 2, 3)
Nameofthevector [2]

```
[1] 2

5. Suppose you'd run this code to make a vector:  
`sentence <- c("I", "am", "not", "learning", "R", "!", "!")` and then you'd run this code: `sentence[c(1,2,4,5)]`  
What would be the output?    [1] "I"        "am"       "learning" "R" 
Does R start counting at 0 or at 1? at 1 
Is this different from Perl? yes, Perl starts counting at 0

6. In chapter 2 of the online R tutorial on Codeschool, you'll learn how to make two kinds of plots. What are they? What functions do you use to make them? In each case, what goes on the x-axis and what on the y-axis? 

The barplot function draws a bar chart with a vector's values. We'll make a new vector for you, and store it in the vesselsSunk variable. > vesselsSunk <- c(4, 5, 1)
> barplot(vesselsSunk)


The plot function takes two vectors, one for X values and one for Y values, and draws a graph of them. Your turn. We'll create a vector with some negative and positive values for you, and store it in the values variable.
We'll also create a second vector with the absolute values of the first, and store it in the absolutes variable.
Try plotting the vectors, with values on the horizontal axis, and absolutes on the vertical axis.
> values <- -10:10
> absolutes <- abs(values)
> plot(values, absolutes)


**Finished!** 

Name: Pilar Lopez

Date:
